modules/social_features/social_event/modules/social_event_addtocal/src/Plugin/SocialAddToCalendarBase.php

```diff
@@ -106,17 +108,17 @@ public function getEventDates(NodeInterface $node) {
     $date_time = [];
 
     // Set formats for event dates.
-    $format = $this->pluginDefinition['dateFormat'];
+    $format = $this->pluginDefinition['dateFormat'] ?? 'Ymd\THis';
     if (date_default_timezone_get() === DateTimeItemInterface::STORAGE_TIMEZONE) {
-      $format = $this->pluginDefinition['utcDateFormat'];
+      $format = $this->pluginDefinition['utcDateFormat'] ?? 'Ymd\THis\Z';
     }
-    $all_day_format = $this->pluginDefinition['allDayFormat'];
+    $all_day_format = $this->pluginDefinition['allDayFormat'] ?? 'Ymd';
 
     // Convert date to correct format.
     // Set dates array.
     if ($all_day) {
       $date_time['start'] = $start_date->format($all_day_format);
-      $end_date->modify($this->pluginDefinition['endDateModification']);
+      $end_date->modify($this->pluginDefinition['endDateModification'] ?? '+ 1 day');
```

Comment:
I'm also a bit lost about what this exactly does. If an end date is empty we return 1 day?